src/activities/ActivityUsageEntity.js

```diff
@@ -200,12 +200,61 @@ export class ActivityUsageEntity extends Entity {
 		await this._setDate(dateValue, Classes.dates.dueDate, 'dueDate');
 	}
 
+	/**
+	 * @returns {boolean} If ActivityAvailabilityDates entity is present
+	 */
+	hasActivityAvailabilityDates() {
+		const dateEntity = this._getSubEntityByClass(Classes.availabilityDates.availabilityDates);
+		return dateEntity !== undefined && dateEntity !== null;
+	}
+
 	/**
 	 * @returns {string} Start date of the activity usage
 	 */
 	startDate() {
-		const startDate = this._getDateSubEntity(Classes.dates.startDate);
-		return startDate && startDate.properties.date;
+		const availabilityStartDate = this._getAvailabilityStartDateEntity();
+
+		if (availabilityStartDate && availabilityStartDate.properties && availabilityStartDate.properties.dateTime) {
+			return availabilityStartDate.properties.dateTime.date;
+		}
+
+		const startDate = this._getSubEntityByClass(Classes.dates.startDate);
+
+		if (!startDate || !startDate.properties) {
+			return;
+		}
+
+		return startDate.properties.date;
+	}
```

Comment:
So I think there may some potential errors that come up from this

The way the entity is added in the backend is if you have date types on the usage, it will add the "activity-availability-dates" entity. However, that entity may not have the "availability-start-date" or "availability-end-date" sub entities if there are no dates set for those.

- [dates entity added here](https://search.d2l.dev/xref/lms/le/activities/D2L.LE.Activities.Extensibility/Api/Hypermedia/Serializers/ActivityUsageDecorator.cs?r=1d2e61c7#141)
- [conditionally adding the date entities here](https://search.d2l.dev/xref/lms/le/activities/D2L.LE.Activities.Extensibility/Api/Hypermedia/ActivitiesSirenSerializers.cs?r=5bec958a&fi=SerializeActivityAvailabilityDates#156-179)

TLDR;
you'll always have a dates entity but you're not guaranteed the date entity

```suggestion
		const availabilityDatesEntity = this._getSubEntityByClass(Classes.availabilityDates.availabilityDates);
		const availabilityStartDate = this._getAvailabilityStartDateEntity();

		if (availabilityDatesEntity ) {
			return availabilityStartDate && availabilityStartDate.properties.dateTime && availabilityStartDate.properties.dateTime.date;
		}

		const startDate = this._getSubEntityByClass(Classes.dates.startDate);

		if (!startDate || !startDate.properties) {
			return;
		}

		return startDate.properties.date;
	}
```

Reply:
The error I'm thinking about could be: 
if you have fad enabled, but you don't have a startDate it won't short circuit and would return the date from the startDate entity

Reply:
Yeah good point, this would fall through in those cases. Will update.